ws/api/transport.go

```diff
@@ -55,6 +57,18 @@ func handshake(svc ws.Service) http.HandlerFunc {
 			return
 		}
 		if err != nil {
+			e, ok := status.FromError(err)
+			if ok {
+				switch e.Code() {
+				case codes.PermissionDenied:
+					w.WriteHeader(http.StatusUnauthorized)
+				case codes.InvalidArgument:
+					w.WriteHeader(http.StatusBadRequest)
```

Comment:
This case is handled in `authorize` function. Request won't be sent if there is no token. If you want, you can update error there.